From: Arnd Bergmann <arnd@arndb.de>
Subject: kexec: include asm/early_ioremap.h
Date: Wed, 23 Apr 2025 18:29:18 +0200

early_memremap() is declared in a header that is only indirectly included
here:

kernel/kexec_handover.c:1116:8: error: call to undeclared function 'early_memremap'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
 1116 |         fdt = early_memremap(fdt_phys, fdt_len);
      |               ^

Link: https://lkml.kernel.org/r/20250423162924.2213664-1-arnd@kernel.org
Fixes: 4bfe738ce89a ("kexec: add KHO support to kexec file loads")
Signed-off-by: Arnd Bergmann <arnd@arndb.de>
Cc: Alexander Graf <graf@amazon.com>
Cc: Baoquan He <bhe@redhat.com>
Cc: Changyuan Lyu <changyuanl@google.com>
Cc: Mike Rapoport <rppt@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 kernel/kexec_handover.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/kernel/kexec_handover.c~x86-add-kho-support-fix
+++ a/kernel/kexec_handover.c
@@ -18,6 +18,9 @@
 #include <linux/memblock.h>
 #include <linux/notifier.h>
 #include <linux/page-isolation.h>
+
+#include <asm/early_ioremap.h>
+
 /*
  * KHO is tightly coupled with mm init and needs access to some of mm
  * internal APIs.
_
